spec/lib/generators/full_generator_test_blobs.rb

```diff
@@ -0,0 +1,193 @@
+# rubocop:disable Metrics/ClassLength, Metrics/MethodLength
+class FullGeneratorTestBlobs
+  def self.simple_action_blob
+    <<~BLOB
+      # frozen_string_literal: true
+
+      class MyAction
+        extend ::LightService::Action
+
+        executed do |ctx|
+        end
+
+        rolled_back do |ctx|
+        end
+      end
+    BLOB
+  end
+
+  def self.simple_action_spec_blob
+    <<~BLOB
+      # frozen_string_literal: true
+
+      require 'rails_helper'
+
+      RSpec.describe MyAction, type: :action do
+        subject { described_class.execute(ctx) }
+
+        let(:ctx) do
+          {
+          }
+        end
+
+        context "when executed" do
+          xit "is expected to be successful" do
+            expect(subject).to be_a_success
+          end
+        end
+      end
+    BLOB
+  end
+
+  # There's some weird whitespace issue which prevents
+  # using HEREDOCS :(
+  def self.advanced_action_blob
+    "# frozen_string_literal: true\n" \
+      "\n" \
+      "module My::Fancy\n" \
+      "  class Action\n" \
+      "    extend ::LightService::Action\n" \
+      "\n" \
+      "    expects  :foo, :bar\n" \
+      "    promises :baz, :qux\n" \
+      "\n" \
+      "    executed do |ctx|\n" \
+      "      foo = ctx.foo\n" \
+      "      bar = ctx.bar\n" \
+      "    end\n" \
+      "  end\n" \
+      "end"
+  end
+
+  def self.advanced_action_spec_blob
```

Comment:
This general strategy is a great way to ensure that the generators are  pumping out the expected static files, but they are slightly brittle when trying to test the dynamic arguments available to the generator.

Would it be at all useful to supply some of those dynamic input values, like `expects_inputs: [:foo, :bar]` and `promises_inputs: [:baz, :qux]` as arguments to this method, and then interpolate those values in the returned blob?

That way tests might better test the dynamic behaviour of the generators.

For example, if something changes in `lib/generators/light_service/templates/action_template.erb`, over here:
```
<%= indent %>expects  <%= @expects.map { |k| ":#{k}" }.join(', ') %>
``` 
having those `expects_inputs` returned by the blob could help detect any regressions if those expressions are inadvertently changed.

Reply:
The additional coverage might not be worth the effort, though.